tests_integration/__tests__/config-resolution.js

```diff
@@ -93,3 +93,55 @@ test("API resolveConfig.sync with file arg and extension override", () => {
     semi: true
   });
 });
+
+test("API resolveConfig with file arg and .editorconfig", () => {
+  const file = path.resolve(
+    path.join(__dirname, "../cli/config/editorconfig/file.js")
+  );
+  return prettier.resolveConfig(file).then(result => {
+    expect(result).toMatchObject({
+      useTabs: true,
+      tabWidth: 8,
+      printWidth: 100
+    });
+  });
+});
+
+test("API resolveConfig.sync with file arg and .editorconfig", () => {
+  const file = path.resolve(
+    path.join(__dirname, "../cli/config/editorconfig/file.js")
+  );
+  expect(prettier.resolveConfig.sync(file)).toMatchObject({
+    useTabs: true,
+    tabWidth: 8,
+    printWidth: 100
+  });
+});
+
+test("API resolveConfig with nested file arg and .editorconfig", () => {
+  const file = path.resolve(
+    path.join(__dirname, "../cli/config/editorconfig/lib/file.js")
+  );
+  return prettier.resolveConfig(file).then(result => {
+    expect(result).toMatchObject({
+      useTabs: false,
+      tabWidth: 2,
+      printWidth: 100
+    });
+  });
+});
+
+test("API resolveConfig.sync with nested file arg and .editorconfig", () => {
+  const file = path.resolve(
+    path.join(__dirname, "../cli/config/editorconfig/lib/file.js")
+  );
+  expect(prettier.resolveConfig.sync(file)).toMatchObject({
+    useTabs: false,
+    tabWidth: 2,
+    printWidth: 100
+  });
+});
+
+test("API clearConfigCache", () => {
+  expect(() => prettier.clearConfigCache()).not.toThrowError();
```

Comment:
This is somewhat out of scope for this PR, but it's an interesting discussion: Are we happy with this test, or do we want to do something like:

1. Read config with cache.
2. Change config file on disk.
3. Read with cache again. Still reports the same as in step 1.
4. Clear cache.
5. Read again. Now the changes from step 2 are picked up.

Reply:
That sounds pretty good to me. We could add another step before 3 (changes in **bold**)

1. Read config with cache.
2. Change config file on disk.
3. **Read without cache. Now the changes from step 2 are picked up.**
3. Read with cache again. Still reports the same as in step 1.
4. Clear cache.
5. Read **from cache** again. Now the changes from step 2 are picked up.

I'd like to do this in a separate PR, though, at least for the non-editorconfig caches (and maybe the editorconfig caches too)